lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -432,18 +432,31 @@ void LocalEnforcer::handle_force_termination_timeout(
 void LocalEnforcer::remove_all_rules_for_termination(
     const std::string& imsi, const std::unique_ptr<SessionState>& session,
     SessionStateUpdateCriteria& uc) {
-  RulesToProcess rules;
-  populate_rules_from_session_to_remove(imsi, session, rules);
-  for (const std::string& static_rule : rules.static_rules) {
+  SessionState::SessionInfo info;
+  session->get_session_info(info);
+
+  for (const std::string& static_rule : info.static_rules) {
     uc.static_rules_to_uninstall.insert(static_rule);
   }
-  for (const PolicyRule& dynamic_rule : rules.dynamic_rules) {
-    uc.dynamic_rules_to_uninstall.insert(dynamic_rule.id());
+  for (const PolicyRule& gx_dynamic_rule : info.dynamic_rules) {
+    uc.dynamic_rules_to_uninstall.insert(gx_dynamic_rule.id());
+  }
+  for (const PolicyRule& gy_dynamic_rule : info.gy_dynamic_rules) {
+    uc.gy_dynamic_rules_to_uninstall.insert(gy_dynamic_rule.id());
   }
+
+  const auto ipv4 = session->get_config().common_context.ue_ipv4();
+  const auto ipv6 = session->get_config().common_context.ue_ipv6();
```

Comment:
can we use `ip_addr` and `ipv6_addr`. We have those two references accross the system 
I was planning to change all of them to `ipv4_addr` and `ipv6_addr` once the ipv6 is done

Reply:
sure ya will change